src/cdk/tree/node.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FocusableOption} from '@angular/cdk/a11y';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  TemplateRef,
+  ViewContainerRef,
+  ViewEncapsulation
+} from '@angular/core';
+import {Subscription} from 'rxjs/Subscription';
+import {FlatNode, NestedNode} from './tree-data';
+
+/** The tree node template */
+export const CDK_TREE_NODE_TEMPLATE = '<ng-content cdkNodeOutlet></ng-content>';
+
+
+/**
+ * Data node definition for the CdkTree.
+ * Captures the node's template and a when predicate that describes when this node should be used.
+ */
+@Directive({
+  selector: '[cdkNodeDef]',
+  inputs: ['when: cdkNodeDefWhen'],
+})
+export class CdkNodeDef<T extends FlatNode|NestedNode> {
+  /**
+   * Function that should return true if this node template should be used for the provided node
+   * data and index. If left undefined, this node will be considered the default node template to
+   * use when no other when functions return true for the data.
+   * For every node, there must be at least one when function that passes or an undefined to
+   * default.
+   */
+  when: (nodeData: T, index: number) => boolean;
+
+  constructor(public template: TemplateRef<any>) {}
+}
+
+
+/**
+ * Tree node for CdkTree. It contains the data in the tree node.
+ */
+// TODO: Role should be group for expandable ndoes
```

Comment:
Can you move the `TODO` next to where `role` is defined? When there's another comment between the JsDoc and the class definition dgeni doesn't connect them